### PR TITLE
Add cco-manual-workload-identity-tp workflow

### PR DESCRIPTION
**Pull request by huangmingxia**
Currently the installation of Azure Workload Identity belongs to TechPreviewNoUpgrade, due to the previous configuration method, TechPreviewNoUpgrade configured in the workflow external, using the cluster-bot to install the cluster can not be configured to configure TechPreviewNoUpgrade, resulting in the failure of the installation, so the environment variable to the workflow level, so as to enable the cluster-bot to create the cluster.
[OCPQE-16001](https://issues.redhat.com/browse/OCPQE-16001)

### CODE REVIEW

**Comment by huangmingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-workload-identity-tp-f14

**Comment by huangmingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-workload-identity-tp-f28-destructive

**Comment by mgahagan73** [comment]
waited for cluster to install and pass health check.

/lgtm

**Review by mgahagan73 — APPROVED** [review]

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-workflow.yaml
Comment (kuiwang02) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-workflow.yaml:9]: @huangmingxia the workflow is changed to TP workflow, I am not sure if it need to indicate it in workflow name because I do not know the rule of workflow name.
if it does not indicate it, I think it will confuse the user.

maybe you need to contact installer team to query it.

Comment (huangmingxia) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-workflow.yaml:9]: @kuiwang02 You are right, you need to indicate it in the workflow.

Comment (huangmingxia) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-workflow.yaml]: @kuiwang02 could you help review again, much thanks.

**Comment by huangmingxia** [comment]
@jianlinliu @yunjiang29 @jhou1 Could you help review this pr? thanks.

**Comment by yunjiang29** [comment]
@huangmingxia is there a way to pass system env to the job triggered by cluster-bot?
/hold

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-workflow.yaml
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-workflow.yaml]: Just like what kuiwang mentioned, if you add the env in the workflow, need a dedicated TP workflow for it to indicate that.

For TP jobs, we encourage mates to set `FEATURE_SET` on job config level, that is why @yunjiang29 is asking if there is some way to set the ENV on cluster-bot side.

**Comment by huangmingxia** [comment]
Hello @yunjiang29 @jianlinliu 

If use cluster-bot to create an environment, it's based on this workflow `cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity`, but I didn't add FEATURE_SET inside the workflow , so cluster-bot doesn't turn on env `FEATURE_SET: TechPreviewNoUpgrade` when calling it to trigger cluster.

Our ci job is workflow+env `FEATURE_SET: TechPreviewNoUpgrade`, 
```
- as: azure-ipi-workload-identity-tp-f14
  cron: 18 7 4,19 * *
  steps:
    cluster_profile: azure-qe
    env:
      BASE_DOMAIN: qe.azure.devcluster.openshift.com
      E2E_RUN_TAGS: '@amd64 and @azure-ipi and @network-ovnkubernetes and not @fips'
      FEATURE_SET: TechPreviewNoUpgrade
      TAG_VERSION: '@4.14'
    test:
    - chain: openshift-e2e-test-qe
    workflow: cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity
```
the cluster created by ci job is enabled with `FEATURE_SET: TechPreviewNoUpgrade`. If we put the FEATURE_SET in the job config level, it can satisfy our normal ci requirements, but the requirements of cluster-bot can't be satisfied at the moment. 

So I think the easiest way is to set the FEATURE_SET at workflow level, which can fulfill our ci requirements and also fulfill the requirements of cluster-bot. After this featuregate is not need, just remove it , and it won't affect the existing ci job and the cluster-bot. If my understanding is wrong, please help to correct it.

**Comment by jianlinliu** [comment]
> After this featuregate is not need, just remove it

No one always remember that, and this time I reviewed the PR, I know the backgroud, next time, maybe other ones hit some problems on the install workflow, do not know the background, now prow ci is a central place that many team is working on, add a `tp` in the install workflow file name can help us quickly debugging and easy-understanding.

**Comment by huangmingxia** [comment]
Hi, @jianlinliu @yunjiang29 update the comments, could you help review it again? thanks.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity-tp/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-tp-workflow.metadata.json
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity-tp/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-tp-workflow.metadata.json:2]: I suggest to create a new folder - `cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/tp/`, so that, in the future, we can easily switch over.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/tp/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-tp-workflow.yaml
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/tp/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-tp-workflow.yaml:5]: you can directly use the provision/deprovision chain in `cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity`, not have to create a copy for the tp workflow.

Comment (huangmingxia) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/tp/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-tp-workflow.yaml]: @jianlinliu  Appreciate it, is it correct now?

**Comment by jianlinliu** [comment]
/lgtm

leave this to @yunjiang29 for more suggestion.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/tp/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-tp-workflow.yaml
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/tp/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-tp-workflow.yaml]: right.

**Comment by yunjiang29** [comment]
@jhou1 can you approve this PR? thanks

**Pull request merged**
